rider-fsharp/src/test/kotlin/lexer/FSharpLexerTest.kt

```diff
@@ -0,0 +1,375 @@
+package lexer
+
+import com.jetbrains.rider.ideaInterop.fileTypes.fsharp.lexer.*
+import com.intellij.lexer.Lexer
+import com.intellij.testFramework.LexerTestCase
+
+class FSharpLexerTest : LexerTestCase() {
+    override fun createLexer(): Lexer {
+        return FSharpLexer()
+    }
+
+    override fun getDirPath(): String? {
+        return null
+    }
+
+    fun testDigit() {
+        doTest("1234567890 1234567890u 1234567890l 0XABCDEFy 0x001100010s 3.0F 0x0000000000000000LF 34742626263193832612536171N",
+                "INT32 ('1234567890')\n" +
+                        "WHITE_SPACE (' ')\n" +
+                        "UINT32 ('1234567890u')\n" +
+                        "WHITE_SPACE (' ')\n" +
+                        "INT32 ('1234567890l')\n" +
+                        "WHITE_SPACE (' ')\n" +
+                        "SBYTE ('0XABCDEFy')\n" +
+                        "WHITE_SPACE (' ')\n" +
+                        "INT16 ('0x001100010s')\n" +
+                        "WHITE_SPACE (' ')\n" +
+                        "IEEE32 ('3.0F')\n" +
+                        "WHITE_SPACE (' ')\n" +
+                        "IEEE64 ('0x0000000000000000LF')\n" +
+                        "WHITE_SPACE (' ')\n" +
+                        "BIGNUM ('34742626263193832612536171N')")
+    }
+
+    fun testString() {
+        doTest("\"STRING\n \\ NEWLINE\n\"",
+                "STRING ('\"STRING\\n \\ NEWLINE\\n\"')")
+    }
+
+    fun testVerbatimString() {
+        doTest("@\"VERBATIM STRING\"",
+                "VERBATIM_STRING ('@\"VERBATIM STRING\"')")
+    }
+
+    fun testByteArray() {
+        doTest("\"ByteArray\"B",
+                "BYTEARRAY ('\"ByteArray\"B')")
+    }
+
+    fun testTripleQuotedString() {
+        doTest("\"\"\"triple-quoted-string\"\"\"",
+                "TRIPLE_QUOTED_STRING ('\"\"\"triple-quoted-string\"\"\"')")
+    }
+
+    fun testSymbolicOperator() {
+        doTest("&&& ||| ?<- @-><-= ?->",
+                "SYMBOLIC_OP ('&&&')\n" +
+                        "WHITE_SPACE (' ')\n" +
+                        "SYMBOLIC_OP ('|||')\n" +
+                        "WHITE_SPACE (' ')\n" +
+                        "SYMBOLIC_OP ('?<-')\n" +
+                        "WHITE_SPACE (' ')\n" +
+                        "SYMBOLIC_OP ('@-><-=')\n" +
+                        "WHITE_SPACE (' ')\n" +
+                        "QMARK ('?')\n" +
+                        "RARROW ('->')")
```

Comment:
Why does `?->` become two tokens? Is there some rule in LexFilter for it? Does it depend on the contexts stack?